src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+        int[] salaries = new int[names.length];
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            LocalDate workDate = LocalDate.parse(parts[0], formatter);
+            String employeeName = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int hourlyRate = Integer.parseInt(parts[3]);
+            if (workDate.isBefore(from) || workDate.isAfter(to)) {
```

Comment:
The current condition `if (workDate.isBefore(from) || workDate.isAfter(to))` excludes the boundary dates. To include the start and end dates, you should adjust the condition to `if (workDate.isBefore(from) || workDate.isAfter(to))` to `if (workDate.isBefore(from) || workDate.isAfter(to))`.